tsdb/querier.go

```diff
@@ -604,17 +604,8 @@ func (p *populateWithDelGenericSeriesIterator) next() bool {
 			p.bufIter.Intervals = p.bufIter.Intervals.Add(interval)
 		}
 	}
-
-	// Re-encode head chunks that are still open (being appended to) or
-	// outside the compacted MaxTime range.
-	// The chunk.Bytes() method is not safe for open chunks hence the re-encoding.
-	// This happens when snapshotting the head block or just fetching chunks from TSDB.
-	//
-	// TODO(codesome): think how to avoid the typecasting to verify when it is head block.
-	_, isSafeChunk := p.currChkMeta.Chunk.(*safeChunk)
-	if len(p.bufIter.Intervals) == 0 && !(isSafeChunk && p.currChkMeta.MaxTime == math.MaxInt64) {
-		// If there is no overlap with deletion intervals AND it's NOT
-		// an "open" head chunk, we can take chunk as it is.
+	if len(p.bufIter.Intervals) == 0 {
+		// If there is no overlap with deletion intervals, we can take chunk as it is.
 		p.currDelIter = nil
 		return true
 	}
```

Comment:
[nit] Below there's this comment:

```
// We don't want the full chunk, or it's potentially still opened, take
// just a part of it.
```

I suggest you to update it, because the case "still opened" is not true anymore.